gnomad_qc/v4/sample_qc/create_sample_qc_metadata_ht_genomes.py

```diff
@@ -0,0 +1,129 @@
+from gnomad_qc.v4.resources.meta import meta as v4_meta
+from gnomad_qc.v4.resources.meta import meta_tsv_path as v4_meta_tsv_path
+from gnomad_qc.v4.resources.sample_qc import hgdp_tgp_meta_updated
+
+logging.basicConfig(format="%(levelname)s (%(name)s %(lineno)s): %(message)s")
+logger = logging.getLogger("Create v4.0 genomes sample QC metadata HT.")
+logger.setLevel(logging.INFO)
+
+
+_meta_versions = {
+    "4.0": TableResource(
+        path="gs://gnomad/v4.0/metadata/exomes/gnomad.exomes.v4.0.sample_qc_metadata.ht"
+    ),
+}
+
+
+def import_updated_annotations(ht: hl.Table, subset_ht: hl.Table) -> hl.Table:
+    """
+    Import updated annotations from subset meta HT to full meta HT.
+
+    :param ht: v3.1 genomes meta HT of genomes
+    :param subset_ht: Updated meta HT of HGDP + TGP subset
+    :return: Updated v4 genomes meta HT
+    """
+    # Some HGDP/TGP samples in v3.1 meta HT have a prefix of "v3.1::" in the key 's',
+    # but the project_meta.sample_id field does not have this prefix. This is used as
+    # a temp key to join the subset meta HT.
+    release_v3 = ht.filter(ht.release).count()
+    ht1 = ht.filter(ht.subsets.hgdp | ht.subsets.tgp)
+    ht1 = ht1.naive_coalesce(5).checkpoint(new_temp_file("v4.0_genomes_meta", "ht"))
+    ht2 = ht.filter(~(ht.subsets.hgdp | ht.subsets.tgp))
+    ht2 = ht2.checkpoint(new_temp_file("v4.0_genomes_meta", "ht"))
+
+    ht1 = ht1.key_by(ht1.project_meta.sample_id)
+    annot = subset_ht[ht1.key]
+    subpop = annot.hgdp_tgp_meta.population
+    freemix = annot.bam_metrics.freemix
+    hard_filters = annot.gnomad_sample_filters.hard_filters
+    hard_filtered = annot.gnomad_sample_filters.hard_filtered
+    release_related = (
+        annot.relatedness_inference.related
+        | annot.gnomad_sample_filters.related_to_nonsubset
+        | annot.gnomad_sample_filters.v4_exome_duplicate
+    )
+    high_quality = annot.gnomad_high_quality
+    release = annot.gnomad_release
+
+    ht1 = ht1.annotate(
+        project_meta=ht1.project_meta.annotate(project_subpop=subpop),
+        bam_metrics=ht1.bam_metrics.annotate(freemix=freemix),
+        sample_filters=ht1.sample_filters.annotate(
+            hard_filters=hard_filters,
+            hard_filtered=hard_filtered,
+            release_related=release_related,
+        ),
+        high_quality=high_quality,
+        release=release,
+    )
+
+    ht1 = ht1.key_by("s").drop("sample_id")
+    ht = ht1.union(ht2)
+
+    # Update gnomAD inferred population 'oth' to be 'remaining'.
+    ht = ht.annotate(
+        population_inference=ht.population_inference.annotate(
+            pop=hl.if_else(
+                ht.population_inference.pop == "oth",
+                "remaining",
+                ht.population_inference.pop,
+            )
+        )
+    )
+    release_v4 = ht.filter(ht.release).count()
+    logger.info(
+        "Number of genome release samples will change from %s in v3.1 to %s in v4.0.",
```

Comment:
```suggestion
        "Number of genome release samples will change from %s in v3.1 to %s in v4",
```

Reply:
it might also be helpful to note as a comment what you are expecting the new number to be. could you also add a check here to make sure that `release_v4` is greater than `release_v3` and throw an error if not?

Reply:
Why do you change all my v4.0 to v4? I had this exchange with Julia that we made everything v4.0 in the freq_ht code.

Reply:
my personal preference is to not hard code v4.0, since that will just mean it is one more thing to update if this code ever needs to be rerun.

that said, Julia has been much closer to the code and what goes into it than I have, so I would defer to her preference here